app/include/zmk/events/battery_state_changed.h

```diff
@@ -14,4 +14,6 @@ struct zmk_battery_state_changed {
     uint8_t state_of_charge;
 };
 
+int32_t zmk_battery_state_of_charge();
```

Comment:
This should be defined in a separate header, not the event header. We keep the event headers "self contained" and focused.

Reply:
added header and changed zmk_battery_state_of_charge(); to uint8_t to match event instead of sensor.